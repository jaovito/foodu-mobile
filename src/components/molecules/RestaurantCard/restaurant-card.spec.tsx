import {fireEvent, render} from '@test-utils';
import {mock} from '../../../test/mocks';
import {RestaurantCard} from '.';

describe('RestaurantCard tests', () => {
  it('should be defined', () => {
    const card = render(
      <RestaurantCard
        name={mock.restaurantCard.name}
        address={mock.restaurantCard.address}
      />,
    );

    expect(card).toBeDefined();
  });

  it('should be able to find the name of the card', async () => {
    const {findByText} = render(
      <RestaurantCard
        name={mock.restaurantCard.name}
        address={mock.restaurantCard.address}
      />,
    );

    const cardName = await findByText(mock.restaurantCard.name);

    expect(cardName).toBeTruthy();
  });

  it('should be able to find the address text of the card', async () => {
    const {findByText} = render(
      <RestaurantCard
        name={mock.restaurantCard.name}
        address={mock.restaurantCard.address}
      />,
    );

    const cardAboutText = await findByText(mock.restaurantCard.address);

    expect(cardAboutText).toBeTruthy();
  });

  it('should be able to click on the card', async () => {
    const onPress = jest.fn();

    const {getByText} = render(
      <RestaurantCard
        name={mock.restaurantCard.name}
        address={mock.restaurantCard.address}
        onPress={onPress}
      />,
    );

    fireEvent.press(getByText(mock.restaurantCard.name));

    expect(onPress).toBeCalled();
  });

  it('should match with snapshot', async () => {
    const onPress = jest.fn();

    const card = render(
      <RestaurantCard
        name={mock.restaurantCard.name}
        address={mock.restaurantCard.address}
        onPress={onPress}
      />,
    ).toJSON();

    expect(card).toMatchSnapshot();
  });
});
