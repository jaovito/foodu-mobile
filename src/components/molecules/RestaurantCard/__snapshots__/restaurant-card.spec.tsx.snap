// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard tests should match with snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderBottomWidth": 1,
      "borderColor": "rgba(0, 0, 0, 0.1)",
      "borderRadius": 16,
      "flexDirection": "row",
      "opacity": 1,
      "paddingBottom": 31,
      "paddingLeft": 31,
      "paddingRight": 31,
      "paddingTop": 31,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/defaultChef.png",
      }
    }
    style={
      [
        {
          "borderRadius": 12,
          "height": 157,
          "marginRight": 55,
          "width": 157,
        },
      ]
    }
  />
  <View
    style={
      [
        {},
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_400Regular",
            "fontSize": 31,
          },
          {},
        ]
      }
    >
      Palpatin
    </Text>
    <Text
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_400Regular",
            "fontSize": 31,
          },
          {},
        ]
      }
    >
      New york
    </Text>
    <RNGestureHandlerButton
      activeOpacity={0.105}
      collapsable={false}
      delayLongPress={600}
      handlerTag={5}
      handlerType="NativeViewGestureHandler"
      onActiveStateChange={[Function]}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FE8C00",
          "borderRadius": 16,
          "justifyContent": "center",
          "marginTop": 16,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "width": "100%",
        }
      }
      underlayColor="black"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "black",
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": undefined,
            "borderRadius": 16,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": undefined,
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <Text
        style={
          [
            {
              "color": "#0D0D0D",
              "fontFamily": "Poppins_400Regular",
              "fontSize": 31,
            },
            {
              "color": "#FFFFFF",
            },
            {
              "fontSize": 16,
              "textAlign": "center",
              "width": 100,
            },
          ]
        }
      >
        general.seeMore
      </Text>
    </RNGestureHandlerButton>
  </View>
</View>
`;
